cmd/controller-bootstrap/command/common.go

```diff
@@ -0,0 +1,210 @@
+			return false, os.MkdirAll(fp, os.ModePerm)
+		}
+		return false, err
+	}
+	if !fi.IsDir() {
+		return false, fmt.Errorf("expected %s to be a directory", fp)
+	}
+	if !isDirWriteable(fp) {
+		return true, fmt.Errorf("%s is not a writeable directory", fp)
+	}
+
+	return true, nil
+}
+
+// isDirWriteable returns true if the supplied directory path is writeable,
+// false otherwise
+func isDirWriteable(fp string) bool {
+	testPath := filepath.Join(fp, "test")
+	f, err := os.Create(testPath)
+	if err != nil {
+		return false
+	}
+	f.Close()
+	os.Remove(testPath)
+	return true
+}
+
+// ensureSDKRepo ensures that we have a git clone'd copy of the aws-sdk-go
+// repository, which we use model JSON files from. Upon successful return of
+// this function, the sdkDir global variable will be set to the directory where
+// the aws-sdk-go is found. It will also optionally fetch all the remote tags
+// and checkout the given tag.
+func ensureSDKRepo(
+	ctx context.Context,
+	cacheDir string,
+	// A boolean instructing ensureSDKRepo whether to fetch the remote tags from
+	// the upstream repository
+	fetchTags bool,
+) error {
+	var err error
+	srcPath := filepath.Join(cacheDir, "src")
+	if err = os.MkdirAll(srcPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	// Clone repository if it doen't exist
+	sdkDir = filepath.Join(srcPath, "aws-sdk-go")
+	if _, err := os.Stat(sdkDir); os.IsNotExist(err) {
+
+		ctx, cancel := context.WithTimeout(ctx, defaultGitCloneTimeout)
+		defer cancel()
+		err = util.CloneRepository(ctx, sdkDir, sdkRepoURL)
+		if err != nil {
+			return fmt.Errorf("canot clone repository: %v", err)
+		}
+	}
+
+	// Fetch all tags
+	if fetchTags {
+		ctx, cancel := context.WithTimeout(ctx, defaultGitFetchTimeout)
+		defer cancel()
+		err = util.FetchRepositoryTags(ctx, sdkDir)
+		if err != nil {
+			return fmt.Errorf("cannot fetch tags: %v", err)
+		}
+	}
+
+	// get sdkVersion and ensure it prefix
+	// TODO: Parse `ack-generate-metadata.yaml` and pass the aws-sdk-go
+	// version here.
+	sdkVersion, err := getSDKVersion("")
+	if err != nil {
+		return err
+	}
+	sdkVersion = ensureSemverPrefix(sdkVersion)
+
+	repo, err := util.LoadRepository(sdkDir)
+	if err != nil {
+		return fmt.Errorf("cannot read local repository: %v", err)
+	}
+
+	// Now checkout the local repository.
+	err = util.CheckoutRepositoryTag(repo, sdkVersion)
+	if err != nil {
+		return fmt.Errorf("cannot checkout tag: %v", err)
+	}
+
+	return err
+}
+
+// ensureSemverPrefix takes a semver string and tries to append the 'v'
+// prefix if it's missing.
+func ensureSemverPrefix(s string) string {
+	// trim all leading 'v' runes (characters)
+	s = strings.TrimLeft(s, "v")
+	return fmt.Sprintf("v%s", s)
+}
+
+// getSDKVersion returns the github.com/aws/aws-sdk-go version to use. It
+// first tries to get the version from the --aws-sdk-go-version flag, then
+// from the ack-generate-metadata.yaml and finally look for the service
+// go.mod controller.
+func getSDKVersion(
```

Comment:
Since aws-sdk-go version is a required field in controller-bootstrap, i do not think you need this method. correct? You can just use that value from command line input.